testbed/tests/window/test_window.py

```diff
@@ -505,6 +688,151 @@ async def test_move_and_resize(second_window, second_window_probe):
+            (WindowState.FULLSCREEN, WindowState.MAXIMIZED),
+            (WindowState.FULLSCREEN, WindowState.PRESENTATION),
+            # Direct switch from PRESENTATION:
+            (WindowState.PRESENTATION, WindowState.NORMAL),
+            (WindowState.PRESENTATION, WindowState.MINIMIZED),
+            (WindowState.PRESENTATION, WindowState.MAXIMIZED),
+            (WindowState.PRESENTATION, WindowState.FULLSCREEN),
+        ],
+    )
+    @pytest.mark.parametrize(
+        "second_window_class, second_window_kwargs",
+        [
+            (
+                toga.MainWindow,
+                dict(title="Secondary Window", position=(200, 150)),
+            )
+        ],
+    )
+    async def test_window_state_direct_change(
+        app,
+        app_probe,
+        initial_state,
+        final_state,
+        second_window,
+        second_window_probe,
+        initial_intermediate_state,
+    ):
+        """Window state can be directly changed to another state."""
+        if (
+            WindowState.MINIMIZED in {initial_state, final_state}
+            and not second_window_probe.supports_minimize
+        ):
+            pytest.xfail(
+                "This backend doesn't reliably support minimized window state."
+            )
+        intermediate_states = [next(initial_intermediate_state)] + random.sample(
+            [state for state in WindowState], len(WindowState)
+        )
+        second_window.toolbar.add(app.cmd1)
+        second_window.content = toga.Box(style=Pack(background_color=CORNFLOWERBLUE))
+        second_window.show()
+
+        # Add delay to ensure windows are visible after animation.
+        await second_window_probe.wait_for_window("Secondary window is visible")
+
+        assert second_window.state == WindowState.NORMAL
+
+        # Set to initial state
+        second_window.state = initial_state
+        # Add delay to ensure windows are visible after animation.
+        await second_window_probe.wait_for_window(
+            f"Secondary window is in {initial_state}",
+            minimize=True if initial_state == WindowState.MINIMIZED else False,
+            full_screen=True if initial_state == WindowState.FULLSCREEN else False,
+        )
+
+        assert second_window.state == initial_state
+
+        # Set to the intermediate states but don't wait for the OS delay.
+        for state in intermediate_states:
+            second_window.state = state
+
+        # Set to final state
+        second_window.state = final_state
+        # Add delay to ensure windows are visible after animation.
+        await second_window_probe.wait_for_window(
+            f"Secondary window is in {final_state}", rapid_state_switching=True
+        )
+
+    @pytest.mark.parametrize(
+        "state",
+        [
+            WindowState.NORMAL,
+            WindowState.MINIMIZED,
+            WindowState.MAXIMIZED,
+            WindowState.FULLSCREEN,
+            WindowState.PRESENTATION,
+        ],
```

Comment:
As with the mobile variant - it's not clear to me how an X->Y test is different from an X->X test.

Reply:
Same reason as https://github.com/beeware/toga/pull/2473#discussion_r1775473697